src/services/players/players.service.ts

```diff
@@ -57,16 +62,40 @@ class PlayersService {
   };
 
   updatePlayerVehicle = (id: string, vehicleId: string, update: PlayerVehicleUpdate) => {
-    // check player exists
-    // check vehicle exists and belongs to player
-    // update vehicle
-    // return updated vehicle
+    const player = this.getPlayer(id);
+    let playerVehicle: PlayerVehicle = vehicles.find(
+      (vehicle) => vehicle.id === vehicleId && vehicle.playerId === player.id,
+    ) as PlayerVehicle;
```

Comment:
i would consider making this a method that throws an error as well similar to `getPlayer`.

the reason being that if someone is trying to access a player or a player's vehicle with the wrong id, we want to throw an error and not let the request move forward.

Reply:
Okay we can talk about how to go about that the next time we meet.

Reply:
sure, quite basic just like the getPlayer
```
  getPlayerVehicle = (playerId: string, vehicleId: string) => {
    const player = this.getPlayer(id);
    const playerVehicleMaybe: undefined | PlayerVehicle = vehicles.find(
      (vehicle) => vehicle.id === vehicleId && vehicle.playerId === playerId,
    );
    if (!playerVehicleMaybe) {
      throw new Error(`Player with id ${playerId} does not own a vehicle with id "${vehicleId}"`);
    }
    return playerVehicleMaybe;
  };
  ```